ls1intum/artemis-android
# `Communication`: Support Attachment Sharing

### Problem Description
This PR introduces a new feature that allows users to upload files and images directly to a conversation.

### Changes
 
**File Upload UI:**

Added buttons for file and image upload in the MarkdownTextField.

**File Upload Feature:**

Supports specific file types, including:

- Images: png, jpg, jpeg, gif, svg
- Documents: pdf, doc, docx, xls, xlsx, ppt, pptx, txt, json, etc.

**Validation:**

Enforced a 5MB file size limit.
Restricted uploads to supported file types.

### Steps for testing
Test the new buttons added to the MarkdownTextField for file and image upload.
Upload a file or image and verify:
Files/images are displayed with appropriate markdown.
Unsupported types and oversized files trigger user-friendly error messages.


### Screenshots

<img width="264" alt="Screenshot 2024-12-07 at 19 50 41" src="https://github.com/user-attachments/assets/187c21f1-6469-4424-8892-675d5e261c0d">
<img width="264" alt="Screenshot 2024-12-07 at 19 50 46" src="https://github.com/user-attachments/assets/6e4d83b1-c0e7-4b4f-bb1b-00a8fc3d3b49">
